add: ollama cloud

vorticalbox opened the pull request:
- [x] I have read [`CONTRIBUTING.md`](https://github.com/charmbracelet/.github/blob/main/CONTRIBUTING.md).
- [x] I have created a discussion that was approved by a maintainer (for new features).


- Set Claude Haiku 4.5 as default small model
- Update Claude Sonnet 4.5 pricing (reduced output costs)
- Remove outdated Claude 3.5 Haiku and Opus models
- Reorder models by relevance

gotjoshua commented:
this looks great! Is there a way i can preview this before the merge?
@vorticalbox 
and @charmcrush  any eta on merge?

vorticalbox commented:
> Hey! Sorry for the delay on this one. In order to get this in we'll need to do some work in [Fantasy](https://github.com/charmbracelet/fantasy) to support Ollama Cloud, since it's not OpenAI or Anthropic compatible. No ETA yet but we'd love to get support for this in.

I already have a pr open in fantasy to support it i just need to update my branch 

https://github.com/charmbracelet/fantasy/pull/63

vorticalbox commented:
> this looks great! Is there a way i can preview this before the merge? @vorticalbox and @charmcrush any eta on merge?

  - Clone both repos: git clone <crush_repo> and git clone <fantasy_repo>.
  - In crush/go.mod, add a replace directive pointing to your local fantasy folder:
    replace charm.land/fantasy => ../fantasy
  - Create providers.json in the Crush repo and paste in the JSON from the provided source.
  - From the Crush repo, run: go crush update-providors path/to/providers.json
  - Start Crush: go run main.go

  After this, Crush will run with fantasy support and can use Ollama Cloud.

vorticalbox committed:
feat: add Ollama Cloud provider configuration

Add provider configuration for Ollama Cloud with the following models:
- DeepSeek V3.1 671B (reasoning)
- GPT-OSS 120B (reasoning)
- GPT-OSS 20B
- Kimi K2 1T (reasoning)
- Qwen3 Coder 480B
- GLM 4.6 (reasoning)
- Minimax M2

All models are currently free (0 cost per token) and support
128K context window with 4K default max tokens.

Requires fantasy provider: charm.land/fantasy/providers/ollamacloud
